agenta-backend/agenta_backend/services/app_manager.py

```diff
@@ -293,7 +294,7 @@ async def add_variant_based_on_image(
     tags: str,
     base_name: str = None,
     config_name: str = "default",
-    template_image: bool = False,
+    deletable: bool = True,
```

Comment:
there is a misunderstanding here. the app_manager should get template_image as input then translate this into deletable. Basically the app_manager is the one who has the business logic: it knows that a template image should not be deleable. the app_router only knows that it is a template image and does not care how the logic is implemented, and the db_manager should only know it is deletable and does not care about what kind of image it is since it concerns itself only about db actions. Each of these has a scope of concerns

Reply:
Thank you for sharing this.